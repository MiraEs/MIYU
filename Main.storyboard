<?xml version="1.0" encoding="UTF-8"?>
<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="14109" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="qd5-jH-GKs">
    <device id="retina4_7" orientation="portrait">
        <adaptation id="fullscreen"/>
    </device>
    <dependencies>
        <deployment identifier="iOS"/>
        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="14088"/>
        <capability name="Safe area layout guides" minToolsVersion="9.0"/>
        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
    </dependencies>
    <scenes>
        <!--View Controller-->
        <scene sceneID="pTX-mo-YIb">
            <objects>
                <viewController id="qd5-jH-GKs" sceneMemberID="viewController">
                    <view key="view" contentMode="scaleToFill" id="VSD-Nj-HQu">
                        <rect key="frame" x="0.0" y="0.0" width="375" height="667"/>
                        <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
                        <color key="backgroundColor" red="0.90101500013384928" green="0.59529076114196156" blue="1" alpha="1" colorSpace="custom" customColorSpace="displayP3"/>
                        <viewLayoutGuide key="safeArea" id="2ns-nX-gjs"/>
                    </view>
                </viewController>
                <placeholder placeholderIdentifier="IBFirstResponder" id="AFQ-mn-LRw" userLabel="First Responder" sceneMemberID="firstResponder"/>
            </objects>
            <point key="canvasLocation" x="74" y="187"/>
        </scene>
    </scenes>
</document>
