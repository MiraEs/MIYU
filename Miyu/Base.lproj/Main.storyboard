<?xml version="1.0" encoding="UTF-8"?>
<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="13771" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="BYZ-38-t0r">
    <device id="retina4_7" orientation="portrait">
        <adaptation id="fullscreen"/>
    </device>
    <dependencies>
        <deployment identifier="iOS"/>
        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="13772"/>
        <capability name="Safe area layout guides" minToolsVersion="9.0"/>
        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
    </dependencies>
    <scenes>
        <!--Login View Controller-->
        <scene sceneID="tne-QT-ifu">
            <objects>
                <viewController id="BYZ-38-t0r" customClass="LoginViewController" customModule="Miyu" customModuleProvider="target" sceneMemberID="viewController">
                    <view key="view" contentMode="scaleToFill" id="8bC-Xf-vdC">
                        <rect key="frame" x="0.0" y="0.0" width="375" height="667"/>
                        <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
                        <subviews>
                            <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" fixedFrame="YES" text="miyu" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="KBZ-oh-NHz">
                                <rect key="frame" x="166" y="57" width="37" height="21"/>
                                <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMaxY="YES"/>
                                <fontDescription key="fontDescription" type="system" pointSize="17"/>
                                <nil key="textColor"/>
                                <nil key="highlightedColor"/>
                            </label>
                            <imageView userInteractionEnabled="NO" contentMode="scaleAspectFill" horizontalHuggingPriority="251" verticalHuggingPriority="251" fixedFrame="YES" image="eye" translatesAutoresizingMaskIntoConstraints="NO" id="qYc-K3-C3b">
                                <rect key="frame" x="67" y="194" width="240" height="128"/>
                                <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMaxY="YES"/>
                            </imageView>
                            <button opaque="NO" contentMode="scaleToFill" fixedFrame="YES" contentHorizontalAlignment="center" contentVerticalAlignment="center" buttonType="roundedRect" lineBreakMode="middleTruncation" translatesAutoresizingMaskIntoConstraints="NO" id="cir-Wt-2rF">
                                <rect key="frame" x="71" y="469" width="38" height="30"/>
                                <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMaxY="YES"/>
                                <state key="normal" title="Login"/>
                            </button>
                            <button opaque="NO" contentMode="scaleToFill" fixedFrame="YES" contentHorizontalAlignment="center" contentVerticalAlignment="center" buttonType="roundedRect" lineBreakMode="middleTruncation" translatesAutoresizingMaskIntoConstraints="NO" id="xzf-Pl-Fei">
                                <rect key="frame" x="231" y="469" width="57" height="30"/>
                                <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMaxY="YES"/>
                                <state key="normal" title="Register"/>
                            </button>
                        </subviews>
                        <color key="backgroundColor" red="1" green="1" blue="1" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
                        <viewLayoutGuide key="safeArea" id="6Tk-OE-BBY"/>
                    </view>
                </viewController>
                <placeholder placeholderIdentifier="IBFirstResponder" id="dkx-z0-nzr" sceneMemberID="firstResponder"/>
            </objects>
        </scene>
    </scenes>
    <resources>
        <image name="eye" width="200" height="200"/>
    </resources>
</document>
