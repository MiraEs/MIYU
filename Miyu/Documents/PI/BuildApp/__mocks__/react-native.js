/* eslint-disable react/no-multi-comp */
import {
	createElement,
	Component,
} from 'react';
import PropTypes from 'prop-types';
const ReactNative = jest.genMockFromModule('react-native');

ReactNative.StyleSheet = {
	create: function create(styles) {
		return styles;
	},
	flatten: function flatten(...styles) {
		return Object.assign({}, ...styles);
	},
};

ReactNative.PixelRatio = {
	get() {
		return 1;
	},
};

ReactNative.Alert = {
	alert: jest.fn(),
};

ReactNative.StatusBar = {
	setBarStyle: jest.fn(),
	setHidden: jest.fn(),
};

ReactNative.Linking = {
	openURL: jest.fn(),
};

ReactNative.NativeModules = {
	PhotoUploader: {},
	SafariViewManager: {},
	UIManager: {
		measure: jest.fn(),
	},
	MarketingCloud: {
		trackAction: jest.fn(),
		trackState: jest.fn(),
	},
	LocalyticsManager: {
		tagScreen: jest.fn(),
		tagEvent: jest.fn(),
		tagCustomerLoggedOut: jest.fn(),
	},
	Instabug: {
		log: jest.fn(),
	},
	BNDevice: {},
	RNApplePayManager: {
		canMakePayments: jest.fn(() => Promise.resolve(true)),
		canMakePaymentsUsingNetworks: jest.fn(() => Promise.resolve(true)),
		paymentRequest: jest.fn(),
		paymentRequestDidFinish: jest.fn(),
		updateShippingMethods: jest.fn(),
		selectShippingMethod: jest.fn(),
		authorizedPayment: jest.fn(),
	},
};

ReactNative.requireNativeComponent = jest.fn();

ReactNative.NetInfo = {
	isConnected: {
		fetch: jest.fn(() => ({ done: jest.fn() })),
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
	},
};

ReactNative.AppState = {
	addEventListener: jest.fn(),
};

ReactNative.CameraRoll = {
	getPhotos: jest.fn(() => Promise.resolve([])),
};

ReactNative.PushNotificationIOS = {
	setApplicationIconBadgeNumber: jest.fn(),
};

ReactNative.Keyboard = {
	dismiss: jest.fn(),
};

ReactNative.findNodeHandle = jest.fn();

ReactNative.ViewPropTypes = {
	style: jest.fn(),
};

ReactNative.NativeEventEmitter = class NativeEventEmitter {
	addListener = jest.fn(() => ({ remove: jest.fn() }));
};

class FlatList extends Component {
	render() {
		return createElement('FlatList', this.props);
	}
}

class SectionList extends Component {
	render() {
		return createElement('SectionList', this.props);
	}
}

class View extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('View', this.props, this.props.children);
	}
}

class ScrollView extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('ScrollView', this.props, this.props.children);
	}
}

class Switch extends Component {
	render() {
		return createElement('Switch');
	}
}

class Text extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Text', this.props, this.props.children);
	}
}

class TextInput extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TextInput', this.props, this.props.children);
	}
}

class TouchableOpacity extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableOpacity', this.props, this.props.children);
	}
}

class TouchableWithoutFeedback extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableWithoutFeedback', this.props, this.props.children);
	}
}

class ToolbarAndroid extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ToolbarAndroid', this.props, this.props.children);
	}
}

class Image extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Image', this.props, this.props.children);
	}
}

class Icon extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('Icon', this.props, this.props.children);
	}
}

class ActivityIndicator extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ActivityIndicator', this.props, this.props.children);
	}
}

ReactNative.Platform = {
	OS: 'ios',
	select: jest.fn((options) => options.ios),
};

ReactNative.InteractionManager = {
	runAfterInteractions: jest.fn(),
};

ReactNative.ActionSheetIOS = {
	showActionSheetWithOptions: jest.fn(),
};

class ListView extends Component {
	DataSource() {
		return {
			cloneWithRows: jest.fn(),
			cloneWithRowsAndSections: jest.fn(),
		};
	}
}

class AppRegistry {
	registerComponent() {}
}

class TouchableHighlight extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableHighlight', this.props, this.props.children);
	}
}
class BackHandler extends Component {
	static addEventListener() {}
	static exitApp() {}
	static removeEventListener() {}
}

class TouchableNativeFeedback extends Component {
	static Ripple() {
	}
}

class Navigator extends Component {
	render() {
	}
}

class RefreshControl extends Component {
	render() {
		return createElement('RefreshControl', this.props);
	}
}

class LayoutAnimation extends Component {
	static configureNext() {
	};

	static Presets() {
		return {
			easeInEaseOut: jest.fn(),
		};
	};
}

class Dimensions {
	static get() {
		return 0;
	}
}

class Animated extends Component {

	static Value() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static ValueXY() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static event() {}
}

class AnimatedView extends Component {
	render() {
		return createElement('AnimatedView');
	}
}

class AnimatedImage extends Component {

}

class AnimatedText extends Component {

}

Animated.View = AnimatedView;

Animated.Image = AnimatedImage;

Animated.Text = AnimatedText;

class PanResponder {
	static create() {
		return {panHandlers: {}};
	}
}

global.__fbBatchedBridgeConfig = {
	remoteModuleConfig: [],
	localModulesConfig: [],
};

ReactNative.View = View;
ReactNative.ScrollView = ScrollView;
ReactNative.FlatList = FlatList;
ReactNative.ListView = ListView;
ReactNative.Text = Text;
ReactNative.TextInput = TextInput;
ReactNative.TouchableOpacity = TouchableOpacity;
ReactNative.TouchableHighlight = TouchableHighlight;
ReactNative.TouchableWithoutFeedback = TouchableWithoutFeedback;
ReactNative.ToolbarAndroid = ToolbarAndroid;
ReactNative.Image = Image;
ReactNative.AppRegistry = AppRegistry;
ReactNative.Icon = Icon;
ReactNative.ActivityIndicator = ActivityIndicator;
ReactNative.BackHandler = BackHandler;
ReactNative.TouchableNativeFeedback = TouchableNativeFeedback;
ReactNative.Navigator = Navigator;
ReactNative.RefreshControl = RefreshControl;
ReactNative.LayoutAnimation = LayoutAnimation;
ReactNative.Dimensions = Dimensions;
ReactNative.Animated = Animated;
ReactNative.PanResponder = PanResponder;
ReactNative.SectionList = SectionList;
ReactNative.Switch = Switch;

module.exports = ReactNative;
