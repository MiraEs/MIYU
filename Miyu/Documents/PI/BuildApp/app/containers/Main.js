import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	AppState,
	NativeModules,
	Linking,
	InteractionManager,
	PushNotificationIOS,
	View,
} from 'react-native';
import { whyDidYouUpdate } from 'why-did-you-update';
import {
	APP_LAUNCH_INFO,
	STORE_CUSTOMER_INFO,
	MORE,
} from '../constants/constants';
import {
	AUTH_STATUS_UNKNOWN,
} from '../constants/Auth';
import TabBar from './TabBar';
import {
	updateDeviceToken,
	registerDeviceToken,
} from '../actions/UserActions';
import environment from '../lib/environment';
import {
	getUnacknowledgedNotificationCount,
	updateNotificationCount,
} from '../actions/NotificationActions';
import {
	handleBranchLink,
	handleLink,
} from '../actions/DeepLinkingActions';
import { showAlert } from '../actions/AlertActions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import EventEmitter from '../lib/eventEmitter';
import styles from '../lib/styles';
import PushNotification from 'react-native-push-notification';
import helpers from '../lib/helpers';
import { refreshCurrentScreen } from '../actions/ErrorActions';
import errorAlertHandler from '../lib/ErrorAlertHandler';
import store from 'react-native-simple-store';
import tracking from '../lib/analytics/tracking';
import branch from 'react-native-branch';
import SimpleStoreHelpers from '../lib/SimpleStoreHelpers';
import ContactUsHelper from '../lib/ContactUsHelper';
import PhoneHelper from '../lib/PhoneHelper';

export class Main extends Component {

	componentWillMount() {
		store.get(APP_LAUNCH_INFO).then((appLaunchInfo) => {
			if (this.props.features.onboarding && (appLaunchInfo === null || appLaunchInfo.isFirstLaunchEver)) {
				this.props.navigator.push('introScreen');
			}
		}).done();

		helpers.setStatusBarStyle('default', true);
		helpers.setStatusBarHidden(false, 'fade');
	}

	componentDidMount() {
		const { handleBranchLink } = this.props.actions;
		if (helpers.isIOS()) {
			PushNotificationIOS.addEventListener('register', this.onRegisterNotifications);
			PushNotificationIOS.addEventListener('notification', this.onRemoteNotification);
		}

		Linking.getInitialURL().then((url) => {
			if (url) {
				this.handleIncomingLink({ url });
			}
		});

		Linking.addEventListener('url', this.handleIncomingLink);

		branch.subscribe((payload) => handleBranchLink(payload));

		AppState.addEventListener('change', this.onAppStateChange);

		store.get(STORE_CUSTOMER_INFO).then((user) => {
			tracking.setCustomerValues(user, this.props.isLoggedIn);
		}).catch(() => {
			tracking.setCustomerValues(null, this.props.isLoggedIn);
		}).done();

		EventEmitter.addListener('showErrorAlert', this.showErrorAlert);
		EventEmitter.addListener('onCallUs', this.onCallUs);

		const { whyDidYouUpdate: whyDidYouUpdateFeature } = this.props.features;
		if (whyDidYouUpdateFeature && __DEV__) {
			whyDidYouUpdate(React);
		}
	}

	componentWillReceiveProps(nextProps) {
		if (helpers.isIOS() && nextProps.isLoggedIn && !this.props.isLoggedIn) {
			PushNotification.checkPermissions(({ alert }) => {
				if (alert) {
					SimpleStoreHelpers.getPushDeviceToken().then((token) => {
						/*
						 * This is a special case where previous logged in user opted for push and they accepted it,
						 * But next logged in user doesn't get asked for push opt-in. So we use the stored device token to
						 * register next logged in user for push.
						 * P.S. We only need to do this on iOS, since we configure push notification after user logs in.
						 */
						if (token) {
							this.onRegisterNotifications(token);
						}
					}).done();
				}
			});
		}

		// Once we receive the auth status, we can route push notifications
		if (this.props.user.status !== nextProps.user.status && nextProps.user.status !== AUTH_STATUS_UNKNOWN) {
			InteractionManager.runAfterInteractions(this.configureNotifications);
		}

		if (nextProps.unauthorizedError && this.props.unauthorizedError !== nextProps.unauthorizedError) {
			EventEmitter.emit('showScreenAlert', {
				message: 'Your session has expired. Please log in again.',
				type: 'error',
				callback: () => {
					this.props.navigation.getNavigator('root').push('loginModal', {
						loginSuccess: () => {
							this.props.navigation.getNavigator('root').pop();
							EventEmitter.emit('screenWillLoad');
						},
					});
				},
			});
		}

		const { whyDidYouUpdate: newWhyDidYouUpdate } = nextProps.features;
		const { whyDidYouUpdate: oldWhyDidYouUpdate } = this.props.features;
		if (newWhyDidYouUpdate !== oldWhyDidYouUpdate && __DEV__) {
			if (newWhyDidYouUpdate) {
				whyDidYouUpdate(React);
			} else {
				// this allows us to effectively turn-off whyDidYouUpdate
				whyDidYouUpdate(React, { exclude: /.*/ });
			}
		}
	}

	componentWillUnmount() {
		if (helpers.isIOS()) {
			PushNotificationIOS.removeEventListener('register', this.onRegisterNotifications);
			PushNotificationIOS.removeEventListener('notification', this.onRemoteNotification);
		}
		Linking.removeEventListener('url', this.handleIncomingLink);

		AppState.removeEventListener('change', this.onAppStateChange);
		EventEmitter.removeListener('showErrorAlert', this.showErrorAlert);
		EventEmitter.removeListener('onCallUs', this.onCallUs);
	}

	configureNotifications = () => {
		if (helpers.isIOS()) {
			PushNotificationIOS.getInitialNotification().then((payload) => {
				if (payload && !this.hasHandledInitialNotification) {
					this.hasHandledInitialNotification = true;
					if (payload._data && payload._data.ll_deep_link_url) {
						// handle deep link from Localytics
						this.handleIncomingLink({
							url: payload._data.ll_deep_link_url,
						});
					} else {
						this.routeNotification(payload);
					}
				}
			});
		} else {
			PushNotification.configure({
				onRegister: ({ token }) => {
					NativeModules.LocalyticsManager.registerPushNotifications();
					this.onRegisterNotifications(token);
				},
				onNotification: (notification) => {
					if (notification.userInteraction) {
						this.props.navigation.performAction(({ tabs }) => tabs('main').jumpToTab(MORE));
						this.props.navigation.getNavigator(MORE).push('singleEvent', {
							fromNotificationClick: true,
							eventId: notification.eventId,
						});
					} else if (notification.foreground) {
						this.props.actions.updateNotificationCount(this.props.notificationCount + 1);
						this.showInAppNotification(notification);
					}
				},
				senderID: environment.gcmSenderId,
				permissions: {
					alert: true,
					badge: true,
					sound: true,
				},
				requestPermissions: true,
			});
		}
	};

	onCallUs = (phone = {}, showSessionInfo = false) => {
		const {
			lastViewedProduct,
			sessionCartId,
			user: { isPro },
		} = this.props;

		if (showSessionInfo) {
			ContactUsHelper.callWithAlert(phone, isPro, sessionCartId, lastViewedProduct);
		} else {
			ContactUsHelper.callUs(phone, isPro);
		}
	};

	routeNotification = (notification) => {
		this.props.navigation.performAction(({ tabs }) => tabs('main').jumpToTab(MORE));
		InteractionManager.runAfterInteractions(() => {
			this.props.navigation.getNavigator(MORE).push('singleEvent', {
				fromNotificationClick: true,
				eventId: notification.eventId || notification._alert && notification._alert.eventId,
			});
		});
	};

	onRegisterNotifications = (token) => {
		tracking.setPushNotificationsEnabled(true);
		const {
			updateDeviceToken,
			registerDeviceToken,
		} = this.props.actions;

		updateDeviceToken(token);
		registerDeviceToken()
			.catch(helpers.noop) // need to decide how we want to handle any errors here
			.done();
	};

	onRemoteNotification = (payload) => {
		if (AppState.currentState === 'active') {
			this.props.actions.updateNotificationCount(this.props.notificationCount + 1);
			this.showInAppNotification(payload);
		} else if (payload && payload._alert && payload._alert.eventId && AppState.currentState !== 'active') {
			const { eventId } = payload._alert;
			if (this.pendingNotificationId === eventId) {
				this.routeNotification(payload);
			} else {
				// When the app is running in the background, onRemoteNotification fires both when the notification is
				// received, and when it is tapped. The payload is identical in each case, so this ensures we will only
				// the event when the user taps the notification.
				this.pendingNotificationId = eventId;
			}
		}
	};

	showInAppNotification = (notification) => {
		if (!(notification._data && notification._data.hasOwnProperty('ll'))) {
			this.props.actions.showAlert('', 'info', {
				text: notification.message || notification._alert && notification._alert.body,
				onPress: () => this.routeNotification(notification),
			}, null, 10000);
		}
	};

	showErrorAlert = (isNetworkError) => {
		const { user } = this.props;
		const phone = PhoneHelper.getPhoneNumberByUserType(user);
		errorAlertHandler.showErrorAlert(this.props.actions.refreshCurrentScreen, isNetworkError, phone);
	};

	handleIncomingLink = ({ url }) => {
		if (url !== null) {
			this.props.actions.handleLink(url);
		}
	};

	onAppStateChange = (appState) => {
		if (appState === 'active') {
			this.props.actions.getUnacknowledgedNotificationCount().catch(helpers.noop).done();
			errorAlertHandler.shouldTryRefresh().then((refresh) => {
				if (refresh) {
					errorAlertHandler.resetShouldTryRefresh();
					this.props.actions.refreshCurrentScreen();
				}
			});
		}
	};

	render() {
		return (
			<View style={styles.elements.flex}>
				<TabBar
					projects={this.props.features.projects}
					shoppingLists={this.props.features.shoppingLists}
				/>
			</View>
		);
	}
}

Main.propTypes = {
	actions: PropTypes.object.isRequired,
	isLoggedIn: PropTypes.bool.isRequired,
	features: PropTypes.object.isRequired,
	notificationCount: PropTypes.number,
	lastViewedProduct: PropTypes.object,
	sessionCartId: PropTypes.number,
	navigator: PropTypes.shape({
		push: PropTypes.func,
	}),
	navigation: PropTypes.shape({
		performAction: PropTypes.func,
		getNavigator: PropTypes.func,
	}),
	unauthorizedError: PropTypes.bool,
	user: PropTypes.object,
};

const mapStateToProps = ({ cartReducer, errorReducer, userReducer, featuresReducer, notificationReducer, productDetailReducer, productsReducer }) => {
	const { lastViewedProductCompositeId } = productDetailReducer;
	let lastViewedProduct = {};
	if (lastViewedProductCompositeId) {
		lastViewedProduct = productsReducer[lastViewedProductCompositeId];
	}
	return {
		lastViewedProduct,
		isLoggedIn: userReducer.isLoggedIn,
		features: featuresReducer.features,
		notificationCount: notificationReducer.notificationCount || 0,
		sessionCartId: cartReducer.cart.sessionCartId,
		unauthorizedError: errorReducer.unauthorized,
		user: userReducer.user,
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		actions: bindActionCreators({
			updateDeviceToken,
			registerDeviceToken,
			getUnacknowledgedNotificationCount,
			updateNotificationCount,
			refreshCurrentScreen,
			handleBranchLink,
			handleLink,
			showAlert,
		}, dispatch),
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(Main);
