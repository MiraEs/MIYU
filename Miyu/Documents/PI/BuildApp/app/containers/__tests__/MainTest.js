'use strict';

jest.mock('../../../app/services/httpClient', () => ({}));
jest.mock('../../../app/store/configStore', () => ({}));
jest.mock('../../../app/lib/analytics/tracking');
jest.mock('BuildNative');
jest.mock('BuildLibrary');
jest.mock('../../../app/components/Form', () => 'Form');
jest.mock('../../../app/components/FormInput', () => 'FormInput');
jest.mock('../../../app/content/Atoms/AtomText@1', () => 'AtomText@1');
jest.mock('../../../app/containers/TabBar', () => 'TabBar');
jest.mock('../../../app/router', () => ({}));
jest.mock('PushNotificationIOS', () => ({
	addEventListener: jest.fn(),
}));

jest.unmock('react-native');
import 'react-native';
jest.mock('Linking', () => ({
	getInitialURL: jest.fn(() => Promise.resolve()),
	addEventListener: jest.fn(),
}));
import { Main } from '../Main';
import React from 'react';

const defaultProps = {
	actions: {
		trackState: jest.fn(),
	},
	features: {},
	isLoggedIn: false,
};

describe('Main component', () => {

	it('should render correctly', () => {
		const tree = require('react-test-renderer').create(
			<Main {...defaultProps} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
