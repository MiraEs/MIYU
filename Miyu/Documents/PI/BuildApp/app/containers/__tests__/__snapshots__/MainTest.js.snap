// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render correctly 1`] = `
<View
  style={
    Object {
      "flexGrow": 1,
    }
  }
>
  <TabBar
    projects={undefined}
    shoppingLists={undefined}
  />
</View>
`;
